chainstate/test-suite/src/tests/orders_tests.rs

```diff
@@ -0,0 +1,1092 @@
+            issue_and_mint_token_from_genesis(&mut rng, &mut tf);
+        let reorg_common_ancestor = tf.best_block_id();
+
+        let ask_amount = Amount::from_atoms(rng.gen_range(10u128..1000));
+        let give_amount = Amount::from_atoms(rng.gen_range(10u128..1000));
+        let order_data = OrderData::new(
+            Destination::AnyoneCanSpend,
+            OutputValue::Coin(ask_amount),
+            OutputValue::TokenV1(token_id, give_amount),
+        );
+
+        let order_id = make_order_id(&tokens_outpoint);
+        let tx = TransactionBuilder::new()
+            .add_input(tokens_outpoint.into(), InputWitness::NoSignature(None))
+            .add_output(TxOutput::AnyoneCanTake(Box::new(order_data.clone())))
+            .build();
+        tf.make_block_builder().add_transaction(tx).build_and_process(&mut rng).unwrap();
+
+        // Fill the order partially
+        let fill_value = OutputValue::Coin(Amount::from_atoms(
+            rng.gen_range(1..ask_amount.into_atoms()),
+        ));
+        let filled_amount = {
+            let db_tx = tf.storage.transaction_ro().unwrap();
+            let orders_db = OrdersAccountingDB::new(&db_tx);
+            orders_accounting::calculate_fill_order(&orders_db, order_id, &fill_value).unwrap()
+        };
+        let left_to_fill = (ask_amount - output_value_amount(&fill_value)).unwrap();
+
+        let tx = TransactionBuilder::new()
+            .add_input(coins_outpoint.into(), InputWitness::NoSignature(None))
+            .add_input(
+                TxInput::AccountCommand(
+                    AccountNonce::new(0),
+                    AccountCommand::FillOrder(order_id, fill_value, Destination::AnyoneCanSpend),
+                ),
+                InputWitness::NoSignature(None),
+            )
+            .add_output(TxOutput::Transfer(
+                OutputValue::TokenV1(token_id, filled_amount),
+                Destination::AnyoneCanSpend,
+            ))
+            .add_output(TxOutput::Transfer(
+                OutputValue::Coin(left_to_fill),
+                Destination::AnyoneCanSpend,
+            ))
+            .build();
+        tf.make_block_builder().add_transaction(tx).build_and_process(&mut rng).unwrap();
+
+        assert_eq!(
+            Some(order_data),
+            tf.chainstate.get_order_data(&order_id).unwrap()
+        );
+        assert_eq!(
+            Some(left_to_fill),
+            tf.chainstate.get_order_ask_balance(&order_id).unwrap()
+        );
+        assert_eq!(
+            Some((give_amount - filled_amount).unwrap()),
+            tf.chainstate.get_order_give_balance(&order_id).unwrap()
+        );
+
+        // Create alternative chain and trigger the reorg
+        let new_best_block = tf.create_chain(&reorg_common_ancestor, 3, &mut rng).unwrap();
+        assert_eq!(tf.best_block_id(), new_best_block);
+
+        assert_eq!(None, tf.chainstate.get_order_data(&order_id).unwrap());
```

Comment:
Maybe add a check showing that the order doesn't exist. I know it's implicit, but it helps to make it explicit.